### ISSUE TITLE
Installation issue: py-tensorflow@2.14.0 +numa

### ISSUE THREAD

**Issue by fgava90** [LLNL/spack#42900]
### Steps to reproduce the issue

```console
$ spack install  py-tensorflow@2.14.0 +numa %gcc@11.3.0
...
ERROR: Config value 'numa' is not defined in any .rc file
```
@adamjstewart @aweits 
py-tensorflow@2.14.0 fails to build when `+numa` . @2.10.0 and @2.12.0 work fine. Haven't tried other versions. 
I suspected an issue with bazel@6.1.0 so i tried with bazel@5.3.0 (same as @2.12.0) but got the same error.

### Error message
```
==> Applied patch /tools/spack/0.21.1/var/spack/repos/builtin/packages/py-tensorflow/null_linker_bin_path.patch
==> py-tensorflow: Executing phase: 'configure'
==> py-tensorflow: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    '/tools/spack/opt/linux-rocky8-x86_64/gcc-11.3.0/bazel/6.1.0-o5i5ymotrl2d2w725d4aadiiqpoeuqjb/bin/bazel' '--nohome_rc' '--nosystem_rc' '--output_user_root=/tmp/spackweoy5szc' 'build' '--color=no' '--jobs=10' '--config=opt' '--verbose_failures' '--config=dynamic_kernels' '--config=cuda' '--config=noaws' '--config=nogcp' '--config=nohdfs' '--config=numa' '--config=v2' '//tensorflow/tools/pip_package:build_pip_package'

1 warning found in build log:
     35    ==> [2024-02-28-10:53:39.946384] FILTER FILE: .tf_configure.bazelrc [replacing "build:opt --host_copt=-march=native"]
     36    ==> py-tensorflow: Executing phase: 'build'
     37    ==> [2024-02-28-10:53:40.001535] '/tools/spack/opt/linux-rocky8-x86_64/gcc-11.3.0/bazel/6.1.0-o5i5ymotrl2d2w725d4aadiiqpoeuqjb/bin/bazel' '--nohome_rc' '--nosystem_rc' '--output_user_root=/tmp/spackweoy5szc' 'build' '--color=no' '--jobs=10' '--config=opt' '--verbose_
           failures' '--config=dynamic_kernels' '--config=cuda' '--config=noaws' '--config=nogcp' '--config=nohdfs' '--config=numa' '--config=v2' '//tensorflow/tools/pip_package:build_pip_package'
     38    $TEST_TMPDIR defined: output root default is '/tmp/spackweoy5szc' and max_idle_secs default is '15'.
     39    Extracting Bazel installation...
     40    Starting local Bazel server and connecting to it...
  >> 41    WARNING: while reading option defaults file '/build/spack/spack-stage/spack-stage-py-tensorflow-2.14.0-7jqxtnbmtst2e5piiufyvmdhki7f3wfl/spack-src/.bazelrc':
     42      invalid command name 'startup:windows'.
     43    ERROR: Config value 'numa' is not defined in any .rc file
```

### Information on your system

* **Spack:** 0.21.1 (e30fedab102f9281a220fb4fae82e3f8c43a82ac)
* **Python:** 3.6.8
* **Platform:** linux-rocky8-x86_64
* **Concretizer:** clingo

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by adamjstewart** [comment]
Don't know much about this one, I would ask the TF developers and ping me on the issue.

**Comment by fgava90** [comment]
@adamjstewart 
This is the issue i opened on TF: https://github.com/tensorflow/tensorflow/issues/63080#issuecomment-1970621536
They suggest to build with clang@16.
Unfortunately several deps fail to build with clang@16...
I doubt that will change anything because the issue is at config level, however i'm not able to get there to prove it

**Comment by adamjstewart** [comment]
Can you try with a modern Python version? I'm not sure how you managed to build with Python 3.8.7 since Spack doesn't allow that with TF 2.14.0.

**Comment by fgava90** [comment]
Thanks for spotting that. I'm doing a million things and got confused. 
I was indeed using python@3.11.6. 
As you can see, i just added a comment in the TF issue i raised.
I suspect `--config=numa` has been removed from 2.14 onwards. 
Not clear to me if on purpose or missing lines in `.bazelrc` are a bug.

**Comment by adamjstewart** [comment]
Closed by #44607
